webui/src/pages/bug/CommentForm.tsx

```diff
@@ -77,12 +78,14 @@ function CommentForm({ bug }: Props) {
     if (issueComment.length > 0) submit();
   };
 
-  function getCloseButton() {
-    return <CloseBugButton bug={bug} disabled={issueComment.length > 0} />;
-  }
-
-  function getReopenButton() {
-    return <ReopenBugButton bug={bug} disabled={issueComment.length > 0} />;
+  function getBugStatusButton() {
+    if (bug.status === 'OPEN' && issueComment.length > 0) {
+      return <CloseBugWithCommentButton bug={bug} comment={issueComment} />;
+    } else if (bug.status === 'OPEN') {
+      return <CloseBugButton bug={bug} />;
+    } else {
+      return <ReopenBugButton bug={bug} disabled={issueComment.length > 0} />;
+    }
```

Comment:
```suggestion
    if (bug.status === 'OPEN' && issueComment.length > 0) {
      return <CloseBugWithCommentButton bug={bug} comment={issueComment} />;
    }
    if (bug.status === 'OPEN') {
      return <CloseBugButton bug={bug} />;
    }
    return <ReopenBugButton bug={bug} disabled={issueComment.length > 0} />;
```

Reply:
Really just a question of style, but you can avoid the `else` if you return anyway. It often makes a cleaner code.